src/routes/v2/authenticated/sites.ts

```diff
@@ -0,0 +1,78 @@
+import autoBind from "auto-bind"
+import express from "express"
+
+import { attachReadRouteHandlerWrapper } from "@middleware/routeHandler"
+
+import UserWithSiteSessionData from "@classes/UserWithSiteSessionData"
+
+import { attachSiteHandler } from "@root/middleware"
+import type { RequestHandler } from "@root/types"
+import type SitesService from "@services/identity/SitesService"
+
+type SitesRouterProps = {
+  sitesService: SitesService
+}
+
+export class SitesRouter {
+  private readonly sitesService
+
+  constructor({ sitesService }: SitesRouterProps) {
+    this.sitesService = sitesService
+    // We need to bind all methods because we don't invoke them from the class directly
+    autoBind(this)
+  }
+
+  getSites: RequestHandler<
+    never,
+    unknown,
+    never,
+    never,
+    { userSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { userSessionData } = res.locals
+    const siteNames = await this.sitesService.getSites(userSessionData)
+    return res.status(200).json({ siteNames })
+  }
+
+  getLastUpdated: RequestHandler<
+    never,
+    unknown,
+    never,
+    { siteName: string },
+    { userSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { siteName } = req.params
```

Comment:
wrong declaration i think - should be first argument. this is confusing because there's 2 kinds of params, path and query.

path parameters are stuff like `/:siteName/something`, where you access by `const {siteName} = req.params` whereas query parameters are `?siteName="abc"`. 

the `RequestHandler` type has `P` (the path param) as its first argument, and this is evidenced here by how you use `req.params` (whereas the type you specified is for `ReqQuery`, which are request parameters)

Reply:
Hmm yeah I think I got confused by the params and query parts yesterday, adjusted in 3658486.